### ISSUE TITLE
Looking for a new maintainer

### ISSUE THREAD

**Issue by FFY00** [pwr-Solaar/Solaar#574]
The state of this project has not been great in the last few years. For the ones that didn't know this repo used to be owned by @pwr. A few years he got inactive, locking up administration priviledges. Because of this it was impossible to add new maintainers and ultimately everything fell into the shoulders of @Lekensteyn, I believe. This was obviously not great as it is too much work for a single person to handle, expecially in an hobby project! Needless to say this took a toll on the project, accumulating a lot of open issues.

Thankfully, arround 4 months ago we were able to reach @pwr and move the repo to the https://github.com/pwr-Solaar organization! After this I steped up as a maintainer, in hope to lift @Lekensteyn of many of his burdens. This was a step in the right diretecion but unfortunetely both me and @Lekensteyn are still not enough to maintain this project properly. So, we are looking for new people to help with that.

If anyone is wiling to become a maintainer you can start by triaging the issues on the bugtracker. Help people debuging issues, try to solve simple issues and send PRs. You can mention me in the issues if you have any technical questions, mainly regaring the protocol, and/or need some hardware tested, since I have a bunch of devices (see https://github.com/libratbag/libratbag/wiki/Devices).

**Comment by rathann** [comment]
Thanks for stepping up. As an owner of a couple of Unifying devices, I recently took over maintenance of `solaar` package in Fedora and updated it to the latest release. It's still not ideal, but far better than what it was a year ago. I can offer some limited help with quality bug reports and triaging as part of my package maintenance work and spread the word that the project needs help.

**Comment by FFY00** [comment]
Thank you! I think what we really need is manpower. Helping out with triaging would be awesome! I can assist on the technical matters of the protocol but I really don't have time to go through the issues. Maybe I'll have some time around Christmas because I get a sort of a break from college.

**Comment by alvgalrus** [comment]
I could lend a hand sometimes, most of the questions are just "I installed 0.9.2 and I'm missing options" anyway... I also have access to three different Logitech wireless mice (G305, MX Anywhere, MX Master).

**Comment by FFY00** [comment]
Other thing new contributors could do: https://github.com/pwr-Solaar/Solaar/issues/582

Figure out which features are available on your device and not yet implemented and try to add support.

**Comment by FFY00** [comment]
Other thing people can help with: #591

Refactor the documentation.

**Comment by pfps** [comment]
I've added closure comments to a few bug reports.
I'm also trying to get receiver c534 to pair better.
Better documentation on current HID++ protocols would help a lot.

**Comment by FFY00** [comment]
Thank you!

>  Better documentation on current HID++ protocols would help a lot.
I will get on that.

**Comment by drupal-daffie** [comment]
Better support for MX Master 3.

**Comment by pfps** [comment]
There are a lot of very old issues.  Most of them should be closed with a comment to try out the current version of solaar.  I can write a fairly nice note to that effect and stick it on the old issues, but I can't close these old issues.
What's the best way to proceed - give me closure rights or have me just add comments and someone else close the issues?

**Comment by FFY00** [comment]
Ping the author to see if the issue is resolved. If there is no response after 1~2 months we close the issues.

**Comment by pfps** [comment]
OK, I'll add something like:

There have been quite a few changes to solaar over the past few years so this issue may have already been addressed.  If you are still using solaar and this issue has not been resolved please download and run the current version from this github repository.  If the problem still persists open a new issue, providing the information requested.

If you are still running solaar you should be able to download and run its current version (in Linux) as follows:

1. Install git on your machine if it is not already installed.  This is generally done via `sudo apt install git` or `sudo dnf install git` 
2. Change to a directory where you want the code.
3. Download solaar via `git clone https://github.com/pwr-Solaar/Solaar.git`
4. Run solaar via `Solaar/bin/solaar`

For more information on installing solaar from github see https://github.com/pwr-Solaar/Solaar/blob/master/docs/installation.md

**Comment by heynnema** [comment]
I commend the supporters and maintainers of solaar. However, updating all old/open issues with this text forces the system to email/notify ALL followers with an email. I've received over 70 emails, and counting. If the email process can be suspending whilst you update, that would save all of us getting more unnecessary emails. Thanks!

**Comment by FFY00** [comment]
Well, I think the damage is done now. The updated issue could have used some triaging but it's all good. Now we just wait for replies and in 1~2 months we go for round 2, closing the stale issues.

**Comment by pfps** [comment]
Not really.  I only did 40 of the very oldest issues.  There are probably about 100 more that are for old versions of solaar (e.g. 0.9.2).  And then the closures will come around as well.
There should be a way of restricting notifications to only those participating in the issue, but that might be against the philosophy of github.  It does appear, however, that setting your watchlevel to "Releases only" will not notify you for messages on an issue you have not participated in, unless you are specifically mentioned.  So watchers who do not want to see the cleanup of old issues should set their watchlevel to this.

**Comment by FFY00** [comment]
Ah, sorry. Working from my phone so I missed that. I don't think we can do anything about the notifications, github will honor the notification level the user set.

**Comment by pfps** [comment]
Those who are watching everything should see this exchange and adjust their watchlevel if they don'https://github.com/pwr-Solaar/Solaart want to see the exchanges to close old issues.  I'll wait for a while before starting the next bunch so that people can make the change if they so desire.  For the record, what I think people have to do is to go to https://github.com/pwr-Solaar/Solaar and click on the Watching menu button, which will probably say Unwatch, and select Releases only.

**Comment by pfps** [comment]
Change your watch level soon if you don't want to see the next bunch of messages about closing old issues.  

I plan on sending out a bunch more messages asking issue reporters whether their issue is still active sometime soon, probably this weekend.  If you are subscribing to all conversations for this project you will be seeing a lot of messages unless you change your watch level to be something other than Watching.

**Comment by pfps** [comment]
As python2.7 is officially deprecated I suggest dropping support for it in solaar.   A bit in the documentation saying so and maybe something added to the installation script is probably needed if so.

**Comment by FFY00** [comment]
Since @pfps stepped up, this is no longer an issue. It would still be great to get more people contributing though. I propose closing this, @pfps what do you think?

**Comment by pfps** [comment]
@FFY00 It would be nice to have testers that would test out changes.  A couple of people are willing, but more would be useful.

**Comment by FFY00** [comment]
Alright, I've renamed the issue.

**Comment by sk8geek** [comment]
Hi, thanks for the continued work on this project. I have MX Keys (and an M350 Pebble mouse), happy to help with testing against that. Would like to help with code but I'm a Java dev so not familiar with Python. I want to understand an issue with my keyboard backlight coming on every few seconds so have spent some time in Wireshark looking through USB packets. I don't think it's Solaar related. Cheers.

**Comment by barolo** [comment]
> Hi, thanks for the continued work on this project. I have MX Keys (and an M350 Pebble mouse), happy to help with testing against that. Would like to help with code but I'm a Java dev so not familiar with Python. I want to understand an issue with my keyboard backlight coming on every few seconds so have spent some time in Wireshark looking through USB packets. I don't think it's Solaar related. Cheers.

@sk8geek you should move your issues to a separate thread, we've been working on MX Keys which is pretty much identical to Craft [ minus its crown ]

**Comment by pfps** [comment]
If you are already watching all conversations for this repository you should be receiving messages about new pull requests.  Checking these messages and testing those PRs that are relevant would be very useful.  I'll try to add a comment about what hardware is needed to test the PR.   Many PRs can be at least partially tested with any hardware.

**Comment by pfps** [comment]
I added a new label to be used in pull requests - Ready for testing.  I'm also adding in comments on what is needed to test the pull request.  Hopefully you have seen a couple of comments - feel free to test out these pull requests.  The easiest way to test a pull request is to fetch it into your clone of Solaar via:
```
git fetch origin pull/XXX/head:pull_XXX
git checkout pull_XXX
```
where XXX is the number of the pull request.
Then you can run the pull request via `bin/solaar`.

**Comment by mschilli87** [comment]
@pfps: I think it would be good to include additional tags for affected devices (or device categories). For example, I wouldn't mind testing the occasional PR that modifies something w.r.t. my Logitech mouse, but having to read all PRs just to find that they affect keyboards only (which I do not own a Solaar-compatible model) would feel like a waste of my time.

In general, I think this repository would benefit from `all devices`, `all keyboards`, `all mice`, `model foo`, `model bar`, etc. tags on issues and PRs as many users are likely affected by only a smaller part of the codebase.

**Comment by pfps** [comment]
I'll see what I can do.  I have been trying to add comments on who should test PRs so you only have to look at the messages about PRs to see what you can test.
